lstchain/calib/camera/calibrator.py

```diff
@@ -128,9 +154,25 @@ def _calibrate_dl1(self, event, telid):
             self.image_extractor.neighbors = camera.neighbor_matrix_where
         charge, pulse_time = self.image_extractor(waveforms)
 
-        event.dl0.event_id = event.r1.event_id
-        event.dl1.tel[telid].image = charge
-        event.dl1.tel[telid].pulse_time = pulse_time + self.mon_data.tel[telid].calibration.time_correction
-
+        # correct time with drs4 correction if available
+        if self.time_corrector:
+            pulse_corr_array = self.time_corrector.get_corr_pulse(event, pulse_time)
+
+        # otherwise use the ff time correction (not drs4 corrected)
+        else:
+            pulse_corr_array = pulse_time + self.mon_data.tel[telid].calibration.time_correction
+
+        # perform the gain selection if the threshold is defined
+        if self.gain_threshold:
+            waveforms, gain_mask = self.gain_selector(event.r1.tel[telid].waveform)
```

Comment:
shouldn't this be calculated on r0?
r1 are in p.e. at this point?

Reply:
The r1 waveform is low-level corrected but still in ADC

Reply:
Isn't this different from CTA data model?

Reply:
Hi @vuillaut,
 thinking gain to this point I agree that  it is better to use the r0 waveform (the fact that it is not low level corrected should not affect the result). I am going to change it.
You are right, our r1 is different than in the ctpipe data model. This is due to the fact that we have a level more in the data (ctapipe r0 is supposed to be already low-level corrected), we should solve this issue in the next (but not in this PR ;-))